notebooks/stitches_dev/nearest_neighboor_matching.R

```diff
@@ -0,0 +1,76 @@
+library(assertthat)
+
+# This function calculates the elucidian distance between the target values (dx and dy) with 
+# and the arhcive values contained in the data frame. It will be used to help select which 
+# of the arhive values best matches the target values. 
+# Args 
+#   fx: a single value of the target fx value
+#   dx: a single value of the target dx value 
+#   data: a data frame of the archive fx and dx values 
+# Return: a dt
+internal_dist <- function(fx, dx, data){
+  
+  #message(data)
+  # calculate the elucian distance between the target point 
+  # and the archive obversations 
+  dist <- unlist(sqrt((data['dx'] - dx)^2 + (data['fx'] - fx)^2))
```

Comment:
@abigailsnyder this is where we will want to save a copy of the value distance vs the rate of change distance. then add it as columns to the out data frame below.